x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/grouping/Categorize.java

```diff
@@ -48,6 +48,12 @@ public class Categorize extends GroupingFunction implements Validatable {
     @FunctionInfo(
         returnType = "keyword",
         description = "Groups text messages into categories of similarly formatted text values.",
+        detailedDescription = """
+            `CATEGORIZE` has the following limitations:
+
+            * can't be used within other expressions
```

Comment:
I wonder if that's too much info, as it's ""technically"" only usable there, and it's documented as a grouping function after all.
Like, yes, we have the logic that "_if the fn is identical to the group, then it's allowed_". But otherwise, it can't appear in the aggs side anyway

Reply:
Fair enough - that's a very narrow edge case that we're not documenting. It's probably fine :)

Reply:
OK, I'll leave this as is